### PR TITLE
[cluster_test] Tag client, faucet images with nightly_tested

### PR DESCRIPTION
**Pull request by ankushagarwal**
# Summary

* We lookup the `upstream_xxxxxx` tag on libra_e2e and tag client and faucet images using it
* Make `fn tag_image()` generic enough so that it can be used in multiple places
* Update `fn tag_tested_image()` so that it tags client, faucet images in addition to libra_e2e
* Create `fn get_upstream_tag()` which fetches the upstream_xxxxxx tag for `libra_e2e:nightly` image
* Refactor the logic of fetching images into `fn get_images()`

# Testing
* Tested using a test tag `ankush_test_tag` and verified that libra_e2e, libra_client and libra_faucet got tagged correctly

Closes https://github.com/libra/libra/issues/1277

### CODE REVIEW

File: testsuite/cluster-test/src/deployment.rs [hunk 1]
```diff
@@ -19,7 +19,9 @@ pub struct DeploymentManager {
 }
 
 const LAST_DEPLOYED_FILE: &str = ".last_deployed_digest";
-const REPOSITORY_NAME: &str = "libra_e2e";
+const VALIDATOR_IMAGE_NAME: &str = "libra_e2e";
```
Comment (andll) [inline testsuite/cluster-test/src/deployment.rs]: Strictly speaking this is repository name, not image name

File: testsuite/cluster-test/src/deployment.rs [hunk 2]
```diff
@@ -117,19 +126,61 @@ impl DeploymentManager {
...
         Ok(())
     }
 
-    fn tag_image(&self, tag: String, hash: String) -> failure::Result<()> {
+    fn get_upstream_tag(&self, image_id: &ImageIdentifier) -> failure::Result<String> {
+        let images = self.get_images(VALIDATOR_IMAGE_NAME, image_id)?;
+        let default_image = &ImageIdentifier::default();
+        let default_tag = &"".to_string();
+        for image in images {
+            let tag = image
```
Comment (andll) [inline testsuite/cluster-test/src/deployment.rs]: I think what you want to do is something like that:
```
let image_id = match image. image_id {
   Some(image_id) => image_id;
   None => continue;
};
let image_tag = match  image_id.image_tag {} {...
```

File: testsuite/cluster-test/src/deployment.rs (hunk 1 above)
Comment (ankushagarwal) [reply testsuite/cluster-test/src/deployment.rs]: Updated to call it `VALIDATOR_IMAGE_REPO`

File: testsuite/cluster-test/src/deployment.rs (hunk 2 above)
Comment (ankushagarwal) [reply testsuite/cluster-test/src/deployment.rs]: Updated

File: testsuite/cluster-test/src/deployment.rs [hunk 3]
```diff
@@ -117,19 +126,62 @@ impl DeploymentManager {
...
         Ok(())
     }
 
-    fn tag_image(&self, tag: String, hash: String) -> failure::Result<()> {
+    fn get_upstream_tag(&self, image_id: &ImageIdentifier) -> failure::Result<String> {
+        let images = self.get_images(VALIDATOR_IMAGE_REPO, image_id)?;
+        let default_image = &ImageIdentifier::default();
+        let default_tag = &"".to_string();
+        for image in images {
+            let image_id = match image.image_id.as_ref() {
```
Comment (andll) [inline testsuite/cluster-test/src/deployment.rs]: Do you need `as_ref()` here? Afaik you should be able to move out of `image` in this context.x

File: testsuite/cluster-test/src/deployment.rs [hunk 4]
```diff
@@ -117,19 +126,62 @@ impl DeploymentManager {
...
+                Some(image_id) => image_id,
+                None => default_image,
+            };
+            let tag = match image_id.image_tag.as_ref() {
+                Some(tag) => tag,
+                None => default_tag,
```
Comment (andll) [inline testsuite/cluster-test/src/deployment.rs]: Why do you need this `default_tag` and `default_image`? You can just `continue;` here

Comment (ankushagarwal) [reply testsuite/cluster-test/src/deployment.rs]: Don't need it. Updated.

File: testsuite/cluster-test/src/deployment.rs (hunk 3 above)
Comment (ankushagarwal) [reply testsuite/cluster-test/src/deployment.rs]: You're right. Updated.\

**Pull request closed without merging**